cli/running/lua/launcher.lua

```diff
@@ -39,22 +39,156 @@ local function check_version(expected)
     return res
 end
 
+-- types of available options
+-- could be comma separated lua types or 'any' if any type is allowed
+--
+-- get_option_from_env() leans on the set of types in use: don't
+-- forget to update it when add a new type or a combination of
+-- types here.
+local template_cfg = {
+    listen              = 'string, number',
+    memtx_memory        = 'number',
+    memtx_min_tuple_size  = 'number',
+    memtx_max_tuple_size  = 'number',
+    slab_alloc_factor   = 'number',
+    work_dir            = 'string',
+    memtx_dir            = 'string',
+    wal_dir             = 'string',
+    vinyl_dir           = 'string',
+    vinyl_memory        = 'number',
+    vinyl_cache               = 'number',
+    vinyl_max_tuple_size      = 'number',
+    vinyl_read_threads        = 'number',
+    vinyl_write_threads       = 'number',
+    vinyl_timeout             = 'number',
+    vinyl_run_count_per_level = 'number',
+    vinyl_run_size_ratio      = 'number',
+    vinyl_range_size          = 'number',
+    vinyl_page_size           = 'number',
+    vinyl_bloom_fpr           = 'number',
+
+    log              = 'string',
+    log_nonblock     = 'boolean',
+    log_level           = 'number',
+    log_format          = 'string',
+    io_collect_interval = 'number',
+    readahead           = 'number',
+    snap_io_rate_limit  = 'number',
+    too_long_threshold  = 'number',
+    wal_mode            = 'string',
+    rows_per_wal        = 'number',
+    wal_max_size        = 'number',
+    wal_dir_rescan_delay= 'number',
+    force_recovery      = 'boolean',
+    replication         = 'string, number, table',
+    instance_uuid       = 'string',
+    replicaset_uuid     = 'string',
+    custom_proc_title   = 'string',
+    pid_file            = 'string',
+    background          = 'boolean',
+    username            = 'string',
+    coredump            = 'boolean',
+    checkpoint_interval = 'number',
+    checkpoint_count    = 'number',
+    read_only           = 'boolean',
+    hot_standby         = 'boolean',
+    worker_pool_threads = 'number',
+    replication_timeout = 'number',
+    replication_sync_lag = 'number',
+    replication_sync_timeout = 'number',
+    replication_connect_timeout = 'number',
+    replication_connect_quorum = 'number',
+    replication_skip_conflict = 'boolean',
+    feedback_enabled      = 'boolean',
+    feedback_host         = 'string',
+    feedback_interval     = 'number',
+    net_msg_max           = 'number',
+}
+
+local module_cfg_type = {
+    -- logging
```

Comment:
Comments must begin with a capital letter and end with a dot.

Reply:
Done